tests/test_developer_tool.py

```diff
@@ -348,7 +348,9 @@ def test_write(self):
     name_backup = project.project_name
 
     # Write and reload.
-    self.assertRaises(securesystemslib.exceptions.Error, project.write)
+    self.assertRaises(tuf.exceptions.UnsignedMetadataError, project.write)
+    project.load_signing_key(project_private_key)
+    project('delegation').load_signing_key(delegation_private_key)
```

Comment:
I think this updated test would be better if we loaded a signing key, did `assertRaises` with `project.write` and then called `project.write(write_partial=True)`.

That way we can test that we continue to support writing partial metadata, even if we _require_ that all metadata be written with at least one signature.

I'd appreciate some input from other maintainers as to whether requiring at least one signature on all metadata is reasonable cc @lukpueh @trishankatdatadog @mnm678

Reply:
I think that it's important to at least warn people when metadata is written without a signature, but I worry that requiring this may impact the ability to sign metadata offline. I believe we want to support the case where metadata is generated on one machine and signed on another.

Reply:
Agree with @mnm678

Reply:
Thanks for the feedback both. For the purposes of this PR it sounds like we want to change this failure to a warning.

I wonder if in future, i.e. our refactor plans, we should have distinct modes of operating the API such that users can't accidentally write metadata without signatures. Instead they'd have to opt-in to generating unsigned/partially signed metadata.

Reply:
Yes, let's argue in future 🙂